osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -234,7 +234,9 @@ protected override void ReloadMappings(IQueryable<RealmKeyBinding> realmKeyBindi
             {
                 base.ReloadMappings(realmKeyBindings);
 
-                KeyBindings = KeyBindings.Where(b => RealmKeyBindingStore.CheckValidForGameplay(b.KeyCombination)).ToList();
+                int keyBindingsUniqueCount = KeyBindings.Select(b => b.KeyCombination).Distinct().Count();
+
+                KeyBindings = (KeyBindings.Count() != keyBindingsUniqueCount) ? DefaultKeyBindings : KeyBindings.Where(b => RealmKeyBindingStore.CheckValidForGameplay(b.KeyCombination)).ToList();
```

Comment:
What is this change for? I don't understand.

Reply:
This is in case of the user already set a binding with repeating `KeyCombination`.

peppy pointed this out in an older PR: #20123

> Keep in mind that this isn't enforcing anything. If the key bindings are already set they are still going to be usable.
> 
> I believe you'll also need some extra logic around this point:
> 
> https://github.com/ppy/osu/blob/31a447fda0532fcf15f8c8251ca890c533386492/osu.Game/Rulesets/UI/RulesetInputManager.cs#L205

Reply:
Ok I think I know what it is for, based from the description:

> If the stored bindings in a ruleset contain repeating key combinations, use the default bindings.

That said this is the completely incorrect approach. Doing things this way means that the keybindings panel in settings will be *actively and silently* lying to the user, because none of the bindings will actually be used in game due to this logic, and there is no indication to the user whatsoever that this is the case.

If this is to happen, it should be a one-time migration and it should be loudly communicated to the user via a notification or something.

Reply:
I made it work. On launch, it checks for repeating key combinations and then posts a notification if it finds repeating key combinations. After that, it sets it to default.

I noticed that osu!catch default bindings have a key combination repeating but the binding they belong to has the same action. So I might want to allow repeating key combinations as long as they have the same action.

btw, I made a container that runs the logic for this and then add it to `OsuGame.cs`. Not sure if it is the correct approach.

Reply:
> I noticed that osu!catch default bindings have a key combination repeating but the binding they belong to has the same action.

Can probably just change the catch default to be a second different key to avoid worrying about this.

> I made it work. On launch, it checks for repeating key combinations and then posts a notification if it finds repeating key combinations. After that, it sets it to default.

Is this on a commit you haven't pushed yet?

Reply:
Just made a commit.

The code is dirty at the moment, I will clean it later.